src/Common/Services/InfrastructureCreationService.cs

```diff
@@ -0,0 +1,185 @@
+
+            // if customerInfrastructure doesn't exists. Can continue to be created.
+            // else check the ignoreIfExists value
+
+            if (currentCustomerInfrastructure != null)
+            {
+                if (ignoreIfExists)
+                {
+                    session.LogInformation($"The Customer Infrastructure with name '{customerInfrastructureName}' already exists but the 'ignoreIfExists' flag has been set. Will be used the current Customer Infrastructure.");
+                }
+                else
+                {
+                    string errorMessage = $"The Customer Infrastructure with name '{customerInfrastructureName}' already exists and cannot be created. If you want to continue, please use the 'ignoreIfExists' flag on the command.";
+                    session.LogError(errorMessage);
+                    throw new Exception(errorMessage);
+                }
+            }
+
+            return currentCustomerInfrastructure;
+        }
+
+        /// <summary>
+        /// Get Customer Infrastructure
+        /// </summary>
+        /// <param name="session">session</param>
+        /// <param name="customerPortalClient">customer portal client</param>
+        /// <param name="customerInfrastructureName">customer infrastructure name</param>
+        /// <param name="throwIfNotExits">throw and exception if not exists</param>
+        /// <returns></returns>
+        public static async Task<CustomerInfrastructure> GetCustomerInfrastructure(ISession session, ICustomerPortalClient customerPortalClient, string customerInfrastructureName, bool throwIfNotExits)
+        {
+            CustomerInfrastructure currentCustomerInfrastructure = null;
+            try
+            {
+                session.LogInformation($"Checking if exists the current Customer Infrastructure with name '{customerInfrastructureName}'.");
+                currentCustomerInfrastructure = await customerPortalClient.GetObjectByName<CustomerInfrastructure>(customerInfrastructureName);
+                session.LogInformation($"Customer Infrastructure with name '{customerInfrastructureName}' exists.");
+            }
+            catch (CmfFaultException ex) when (ex.Code?.Name == Cmf.Foundation.Common.CmfExceptionType.Db20001.ToString() && ex.Message.Contains("was not found in the system"))
+            {
+                // when was not found
+
+                string errorMessage = $"The Customer Infrastructure with name '{customerInfrastructureName}' doesn't exists.";
+                if (throwIfNotExits)
+                {
+                    session.LogError(errorMessage);
+                    throw new Exception(errorMessage);
+                }
+
+                session.LogInformation(errorMessage);
+            }
+            // other exceptions will throw!
+
+            return currentCustomerInfrastructure;
+        }
```

Comment:
In the beegining this was the same method but I split because we can have the situation where we just want to check and if doesn'tt exists throw (like is happening in the call of NewInfrastructureFromTemplateHandler.cs). I can try merge, but I think we will add complexity and difficult to read the code.

Reply:
done :)